samples/sample-expunge-vpc/src/lambda_vpc/lambda_function.py

```diff
@@ -153,50 +177,56 @@ def get_regions(client):
 def create(event, context):
     logger.info("Stack creation therefore default VPCs are to be removed")
 
-    client = create_ec2_client(region_name)
+    client = create_ec2_client(REGION_NAME)
     regions = get_regions(client)
     for region in regions['Regions']:
-        logger.info(f"Creating ec2 client in {region['RegionName']} region")
+        logger.info("Creating ec2 client in %s region", region['RegionName'])
         logger.info(
-            "Calling 'remove_default_vpc' function to remove default VPC and associated resources within the region")
+            "Calling 'remove_default_vpc' function to remove default VPC and "
+            "associated resources within the region",
+        )
         ec2_client = create_ec2_client(region_name=region['RegionName'])
         remove_default_vpc(ec2_client)
         logger.info('~' * 72)
 
     # Items stored in helper.Data will be saved
     # as outputs in your resource in CloudFormation
     helper.Data.update({})
-    return generate_dummy_resource_id(event)  # This is the Physical resource of your ID
+    # This is the Physical resource of your ID:
+    return generate_dummy_resource_id(event)
 
 
 @helper.update
 def update(event, context):
     logger.info("Stack update therefore no real changes required on resources")
-    return generate_dummy_resource_id(event)  # This is the Physical resource of your ID
+    # This is the Physical resource of your ID:
+    return generate_dummy_resource_id(event)
 
 
 @helper.delete
 def delete(event, context):
     logger.info("Stack deletion therefore default VPCs are to be recreated")
 
-    client = create_ec2_client(region_name)
+    client = create_ec2_client(REGION_NAME)
     regions = get_regions(client)
     for region in regions['Regions']:
-        logger.info(f"Creating ec2 client in {region['RegionName']} region")
+        logger.info("Creating ec2 client in %s region", region['RegionName'])
         ec2_client = create_ec2_client(region_name=region['RegionName'])
-        vpcs = ec2_client.describe_vpcs()
-        if len(vpcs['Vpcs']) == 0:
+        vpc_paginator = ec2_client.get_paginator('describe_vpcs')
+        vpc_pages = vpc_paginator.paginate()
+        default_vpc_found = False
+        for vpcs in vpc_pages:
+            for vpc in vpcs['Vpcs']:
+                if vpc['IsDefault']:
+                    default_vpc_found = True
+        if not default_vpc_found:
             logger.info("Creating default VPC in region")
             ec2_client.create_default_vpc()
-        else:
-            for vpc in vpcs['Vpcs']:
-                if not vpc['IsDefault']:
-                    logger.info("Creating default VPC in region")
-                    ec2_client.create_default_vpc()
 
         logger.info('~' * 72)
```

Comment:
Can we get rid of this?

Reply:
When it iterates between accounts, the line is added to the logs to indicate that it switches from one account to another. So it is easier to trace. I looked into that line as well before, but kept it as I think it adds value.